src/types/typedefs.ts

```diff
@@ -22,21 +22,48 @@ export type Database = Technology & {
   schema: string; // as code snippet
 };
 
+// export type ProjectData = {
+//   id: string;
+//   idea: string;
+//   title: string;
+//   summary: string;
+//   frontend: {
+//     todoList: string[];
+//     framework: Technology;
+//     colorScheme: ColorScheme;
+//   };
+
+//   backend: {
+//     todoList: string[];
+//     framework: Technology;
+//     database: Database;
+//   };
+//   createdAt: number;
+// };
+
 export type ProjectData = {
   id: string;
+  userId: string;
   idea: string;
   title: string;
   summary: string;
+  forontendId: string;
```

Comment:
you have a typo :)
also:
1- why do we need IDs in this type definition? that would mean we need to generate IDs as soon as we get the data from  AI.
2- in case we really need them, why not putting 'id' inside the 'frontend' object, instead of at the same level?